protostar-rust/src/main.rs

```diff
@@ -1,30 +1,37 @@
-use anyhow::{Context, Result};
+use anyhow::{anyhow, Context, Result};
 use camino::Utf8PathBuf;
 use clap::Parser;
+use include_dir::{include_dir, Dir};
 use scarb_metadata::MetadataCommand;
+use std::path::PathBuf;
+use tempfile::{tempdir, TempDir};
 
 use rust_test_runner::pretty_printing;
 use rust_test_runner::run_test_runner;
 
+static CORELIB: Dir = include_dir!("../cairo/corelib/src");
+
 #[derive(Parser, Debug)]
 struct Args {
     test_filter: Option<String>,
-    // TODO #1997 this is a temporary solution for tests to work, this argument
-    //  should be detected automatically
-    #[arg(short, long)]
-    corelib_path: Option<String>,
+}
+
+fn load_corelib() -> Result<TempDir> {
+    let tmp_dir = tempdir()?;
+    CORELIB
+        .extract(&tmp_dir)
+        .expect("Failed to copy corelib to temporary directory");
+    Ok(tmp_dir)
 }
 
 fn main_execution() -> Result<()> {
-    let args = Args::parse();
+    let _args = Args::parse();
 
     // TODO #1997
-    let corelib = match args.corelib_path {
-        Some(corelib) => Utf8PathBuf::from(corelib),
-        None => Utf8PathBuf::from("../../cairo/corelib/src")
-            .canonicalize_utf8()
-            .context("Failed to resolve corelib path")?,
-    };
+    let corelib_dir = load_corelib()?;
```

Comment:
Why is `load_corelib` a separate function? This code isn't reused and is not even big. I think it should be inlined here.

Reply:
I am not sure tbh, it abstracts the operation nicely

Reply:
It can be brought down to literally one instruction:
```
CORELIB
        .extract(&tempdir()?)
        .expect("Failed to copy corelib to temporary directory");
```

Reply:
Yea but looking at it i am not sure what it does at first glance and function names it

Reply:
LoL, the error message tells you exactly what it does:

> Failed to copy corelib to temporary directory

which means that what it does is: "copy corelib to temporary directory". It's even more descriptive than the function name "load_corelib" (btw isn't this name misleading? "load corelib" does not indicate that we copy it to a tmp dir, it might also mean that we load this into the memory or something which brings confusion).

> I would stick with a function, this way we reduce cognitive complexity imo

Extracting single instructions into separate once-used functions looks like an anti-pattern to me. This leads to a code I'd not wish to maintain. Also, as I mentioned above, the name of this function is not quite compatible with the error message and with what's actually happening and it is misleading.

Long story short, I know it's 2v1 guys and usually in such a case I back off right away but here I find your arguments not super legit, sorry 🤷 At the end of the day I'm only an advisor here, the author may do as he wishes.

Reply:
I'd rather have multiple small functions in the codebase that make it readable. From the point of person doing something in context of `fn main`, how exactly is corelib loaded is not important at all.